src/fugw/datasets/surf_geometry.py

```diff
@@ -0,0 +1,165 @@
+import gdist
+import numpy as np
+
+from typing import Tuple
+
+from joblib import Memory
+from nilearn import surface, datasets
+from scipy.spatial import distance_matrix
+
+from fugw.scripts import coarse_to_fine, lmds
```

Comment:
I would argue that coarse-to-fine should be moved under mappings, since it defines a fit() method. The rest (lmds, ward algorithm...) would be better off in `fugw.utils`. Maybe let's do this refactoring in another PR? This one is already pretty heavy.

Reply:
Yep, makes sense.